src/main/java/uk/gov/hmcts/reform/blobrouter/services/BlobContentExtractor.java

```diff
@@ -0,0 +1,39 @@
+package uk.gov.hmcts.reform.blobrouter.services;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.blobrouter.config.TargetStorageAccount;
+import uk.gov.hmcts.reform.blobrouter.exceptions.InvalidZipArchiveException;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.Objects;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import static com.google.common.io.ByteStreams.toByteArray;
+import static uk.gov.hmcts.reform.blobrouter.config.TargetStorageAccount.CRIME;
+import static uk.gov.hmcts.reform.blobrouter.util.zipverification.ZipVerifiers.ENVELOPE;
+
+@Component
+public class BlobContentExtractor {
+
+    public byte[] getContentToUpload(byte[] blobContent, TargetStorageAccount targetAccount) throws IOException {
+        if (targetAccount == CRIME) {
+            try (var zipStream = new ZipInputStream(new ByteArrayInputStream(blobContent))) {
+                ZipEntry entry;
+
+                while ((entry = zipStream.getNextEntry()) != null) {
+                    if (Objects.equals(entry.getName(), ENVELOPE)) {
+                        return toByteArray(zipStream);
+                    }
+                }
+
+                throw new InvalidZipArchiveException(
+                    String.format("ZIP file doesn't contain the required %s entry", ENVELOPE)
+                );
+            }
+        } else {
+           return blobContent;
+        }
+    }
```

Comment:
maybe we could if/else in method caller instead? this is content extractor afterall

Reply:
My goal was to remove the logic around whether to extract (plus how) or not to not test everything twice for both CRIME and BULKSCAN in the unit test for processor.

How about a different name for this new class then?

Reply:
fair enough. we tend to struggle with names. right now can't think of one, maybe it'll kick in later :)

Reply:
`TargetContentPreparator`?

Reply:
:bowing_man:

Reply:
`BlobContentHandler` ?